src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -370,12 +402,42 @@ private void AddHash()
                 // We can exclude more in the future.
 
                 StringBuilder.Append("/* HASH ");
-                Parameters.AppendHash(StringBuilder);
-                Parameters.AppendHashedParameterNames(StringBuilder);
+                if (forSmartV2Include)
```

Comment:
One issue that HSE encountered was the extensive compile time for SMART queries, which necessitated plan caching.  I assume that with caching enabled, this hash won't be used, and caching will still take affect, correct?.  However, for other customers do we anticipate that more items in the hash will result in a higher likelihood of plan recalculation?  I realize this may be significantly mitigated by the simplified Compartment CTEs, which hopefully reduces plan calculation time overall.  Did we do any kind of testing or analysis of plan compilation times with these new items in the hash?